tests/integration/ceph_mgr_test.go

```diff
@@ -122,19 +114,73 @@ func (s *CephMgrSuite) execute(command []string) (error, string) {
 	return s.installer.Execute("ceph", orchCommand, s.namespace)
 }
 
+func (s *CephMgrSuite) enableOrchestratorModule() error {
+	logger.Info("Enabling Rook orchestrator module: <ceph mgr module enable rook --force>")
+	err, output := s.installer.Execute("ceph", []string{"mgr", "module", "enable", "rook", "--force"}, s.namespace)
+	logger.Infof("output: %s", output)
+	if err != nil {
+		logger.Infof("Not possible to enable rook orchestrator module: %q", err)
+		return err
+	}
+	logger.Info("Setting orchestrator backend to Rook .... <ceph orch set backend rook>")
+	err, output = s.execute([]string{"set", "backend", "rook"})
+	logger.Infof("output: %s", output)
+	if err != nil {
+		logger.Infof("Not possible to set rook as backend orchestrator module: %q", err)
+	}
+	return err
+}
+
 func (s *CephMgrSuite) waitForOrchestrationModule() {
 	var err error
+
+	// Status struct
+	type orchStatus struct {
+		Available bool   `json:"available"`
+		Backend   string `json:"backend"`
+	}
+
+	err = s.enableOrchestratorModule()
+	if err != nil {
+		logger.Error("First attempt: Error trying to set Rook orchestrator module")
+	}
+
 	for timeout := 0; timeout < 30; timeout++ {
-		err, output := s.execute([]string{"status"})
+		logger.Info("Waiting for rook orchestrator module enabled and ready ...")
+		err, output := s.execute([]string{"status", "--format", "json"})
 		logger.Infof("%s", output)
 		if err == nil {
 			logger.Info("Rook Toolbox ready to execute commands")
-			return
+
+			// Get status information
+			bytes := []byte(output)
+			var status orchStatus
+			err := json.Unmarshal(bytes, &status)
+			if err != nil {
+				logger.Error("Error getting ceph orch status")
+				continue
+			}
+
+			if status.Backend != "rook" {
+				err = s.enableOrchestratorModule()
```

Comment:
Since we expect the backend to be set by the operator, how about if we add an `assert.Fail()` here so we know the operator had an issue? But as discussed we will continue with the test enabling it for test reliability

Reply:
As the problem is usually that ceph toolbox is not ready, i have simplified the code to only enable and set as backend the rook orchestrator if this is not set. 
Now it runs fast and smoother . Thanks for the suggestions